### DISCUSSION TITLE
Support Alpine and use SparrowCI for auto testing

### DISCUSSION THREAD

**Discussion by melezhik** [cirospaciari/socketify.py#26] [Q&A] [answered]
Hi! I am building a free CI service with support of Python projects. I noticed that your GH CI only has Ubuntu ( for Linux ). Are you interested  In testing under other Linux distributions? SparrowCI does out of the box Debian/Ubuntu/Arch Linux / Alpine. An example of Python project ( pyinfra) build - https://ci.sparrowhub.io/report/1542

**Comment by cirospaciari** [comment]
Should work fine with the same binary in arch, without need of other build, I will do some tests and if needed will add new build too.

I use Debian and it's work fine, I have a docker file for alpine I will check  if last version if is running fine too, and will check Arch too.

If you find some bug with your/other distro feel free to open an issue

**Comment by cirospaciari** [comment, marked as answer]
@melezhik I tested with Debian and Arch and other distros too, no need to new type of build.
BUT i need an alpine build, so i will create an issue to support alpine  musl ABI :) https://github.com/cirospaciari/socketify.py/issues/27

I will rename this discussion for "Support Alpine", if you have any other distros with problems feel free to open another discussion, issue or even an PR :zap:

And i already have an Issue for supporting Windows on ARM and Linux on ARM here https://github.com/cirospaciari/socketify.py/issues/19
Thank!

**Reply by melezhik** [reply]
@cirospaciari the point of this discussion is you don’t have to run tests under every Linux distribution manually , you can use SparrowCI that supports many Linux distributions unlike GH action that only does Ubuntu out of the box . I have no doubt you can always check manually when you need, but the idea is to have CI for that

**Reply by cirospaciari** [reply]
@melezhik mostly of this project is pure python with FFI, and the native part it self is already an very well tested (uWebSockets and Bun uses the same lib), so i not too worried about do CI testing using Sparrow at the moment, but more worried in with the goals it self, and after this add more automation in testing and  maybe use Sparrow to it, Thanks for indication SparrowCI i will take a deeper look!

**Reply by cirospaciari** [reply]
> Comment options
> ####
> [ melezhik](/melezhik)

I opened an Issue to take a look on it https://github.com/cirospaciari/socketify.py/issues/28 it will be not get forget thanks again for you recommendation its means a lot!
